### PR TITLE
Include Dockerfile for Serverpod Mini projects

### PR DESCRIPTION
**Pull request by klkucaj**
Closes: [#2605](https://github.com/serverpod/serverpod/issues/2605)

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [ ] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

### CODE REVIEW

**Comment by Isakdl** [comment]
The current Dockerfile does not work with mini as we are copying folders that do not exist.

Docker doesn't really support doing optional copies, but there is a hack to do it anyway.

```
# Hack to optionally copy config, web and migrations
COPY --from=build /app/bin/server /app/confi[g]/ /app/we[b]/ /app/migration[s]/ /app/
```

Anyway the above needs to replace the current copy commands the Dockerfile for it to work with mini.

**Comment by klkucaj** [comment]
@Isakdl is this okay
https://github.com/serverpod/serverpod/blob/334d7f43a437bab47c3924ce447ce7a50a71a4a7/templates/serverpod_templates/projectname_server/Dockerfile#L18-L19

**Comment by Isakdl** [comment]
No unfortunately not, at least one file / folder must be copied so you have to put all of them on the same line together with the binary that is guaranteed to be there.

**Comment by klkucaj** [comment]
@Isakdl we already talked about this but i am just going to leave this screenshot as proof that it works:
Screenshot 2024-09-25 at 2 12 58 PM

File: templates/serverpod_templates/projectname_server/Dockerfile [hunk 1]
```diff
@@ -15,8 +15,9 @@ ENV role=monolith
 
 COPY --from=build /runtime/ /
 COPY --from=build /app/bin/server server
-COPY --from=build /app/config/ config/
-COPY --from=build /app/web/ web/
+COPY --from=build /app/confi[g]/ config/
+COPY --from=build /app/we[b]/ web/
+
```
Comment (Isakdl) [inline templates/serverpod_templates/projectname_server/Dockerfile:20]: Suggestion: Add the migrations folder too!

Comment (klkucaj) [reply templates/serverpod_templates/projectname_server/Dockerfile:20]: @Isakdl we dont have that currently:

https://github.com/serverpod/serverpod/blob/97d72a2ac377cc81a82a2e9f2ef6a82717283e12/templates/serverpod_templates/projectname_server_upgrade/Dockerfile#L1-L25

Should i add it still?

Comment (klkucaj) [reply templates/serverpod_templates/projectname_server/Dockerfile:21]: @Isakdl done!

**Pull request merged**